assets/app/scripts/directives/oscObjectDescriber.js

```diff
@@ -0,0 +1,122 @@
+angular.module('openshiftConsole')
+  .directive('oscObjectDescriber', function(ObjectDescriber) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: 'views/directives/osc-object-describer.html',
+      link: function(scope, elem, attrs) {
+        var callback = ObjectDescriber.onResourceChanged(function(resource, kind) {
+          scope.$apply(function() {
+            scope.kind = kind;
+            scope.resource = resource;
+          });
+        });
+        scope.$on('$destroy', function() {
+          ObjectDescriber.removeResourceChangedCallback(callback);
+        });    
+      }
+    };
+  })
+  .directive('oscObject', function(ObjectDescriber) {
+    return {
+      restrict: 'AC',
+      scope: {
+        resource: '=',
+        kind: '@'
+      },
+      link: function(scope, elem, attrs) {
+        $(elem).on("click.oscobject", function() {
```

Comment:
i left it being registered this way so that if we had other directives adding/removing click handlers then they would never affect these.  Basically trying to get into the habit of using namespaced click handlers.